libc/math/frac.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ *
+ * HEADER:
+ *    - math.h
+ *
+ * TESTED:
+ *    - modf(), fmod()
+ *    - ceil, floor()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <math.h>
+#include <limits.h>
+#include <float.h>
+#include <time.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "common.h"
+#include <unity_fixture.h>
+
+static double test_ceilAndFloor(double x, int type)
+{
+	double powTwo, frac = x;
+
+	if (frac < 0.0) {
+		frac = -frac;
+	}
+
+	if (frac > pow(2.0, test_common.maxPowTwo)) {
+		return x;
+	}
+
+	for (powTwo = pow(2.0, test_common.maxPowTwo); powTwo >= 2.0; powTwo /= 2.0) {
+		while (frac > powTwo) {
+			frac -= powTwo;
+		}
+	}
+
+	while (frac >= 1.0) {
+		frac -= 1.0;
+	}
```

Comment:
Yeah, it uses division, which is slow, but your code too (in for loop).

I guess you want to avoid using `frexp()`?
```suggestion
	while (frac >= 1.0) {
		frac /= 2.0;
	}
```